collated/test/A0139052L.md

```diff
@@ -0,0 +1,235 @@
+     * @param categories the categories in order in which to delete from
+     * @param currentList A copy of the current list of tasks (before deletion).     
+     */
+    private void assertDeleteSuccess(int[] targetIndexes, char[] categories, final TestTaskList currentList) {
+        
+        StringBuilder commandText = new StringBuilder("delete ");
+        StringBuilder resultMessage = new StringBuilder();
+        
+        for (int i = 0; i < targetIndexes.length; i++) {
+            
+            TestTask taskToDelete = currentList.getTaskFromList(targetIndexes[i] - 1, categories[i]); //-1 because array uses zero indexing
+            currentList.removeTaskFromList(targetIndexes[i] - 1, categories[i]);
+            
+            commandText.append(categories[i]);
+            commandText.append(targetIndexes[i] + " ");
+            
+            int categoryIndex = TaskUtil.getCategoryIndex(categories[i]);
+            resultMessage.append(String.format(MESSAGE_DELETE_TASK_SUCCESS, Task.CATEGORIES[categoryIndex], taskToDelete));
+        }
+        
+        commandBox.runCommand(commandText.toString());
+                
+        //confirm the list now contains all previous persons except the deleted person
+        assertTrue(currentList.isListMatching(taskListPanel));
+
+        //confirm the result message is correct
+        assertResultMessage(resultMessage.toString());
+    }
+    
+}
+```
+###### /java/guitests/DoneCommandTest.java
+``` java
+    /**
+     * 
+     * Runs the done command to mark the tasks at specified indexes as done and confirms the result is correct 
+     * @param targetIndexes the indexes in order to which to mark as done from.
+     * @param categories the categories in order in which to mark as done from.
+     * @param currentList A copy of the current list of tasks (before being marked as done).     
+     */
+    private void assertMarkAsDoneSuccess(int[] targetIndexes, char[] categories, final TestTaskList currentList) {
+               
+        StringBuilder commandText = new StringBuilder("done ");
+        StringBuilder resultMessage = new StringBuilder();
+        TestTask[] markedTasks = new TestTask[targetIndexes.length];
+        
+        for (int i = 0; i < targetIndexes.length; i++) {
+            
+            TestTask taskToMark = currentList.getTaskFromList(targetIndexes[i] - 1, categories[i]); //-1 because array uses zero indexing            
+            currentList.markTaskAsDoneInList(targetIndexes[i] - 1, categories[i], taskToMark);
+                        
+            commandText.append(categories[i]);
+            commandText.append(targetIndexes[i] + " ");
+            
+            int categoryIndex = TaskUtil.getCategoryIndex(categories[i]);
+            resultMessage.append(String.format(MESSAGE_MARK_TASK_AS_DONE_SUCCESS, Task.CATEGORIES[categoryIndex], taskToMark));
+            
+            markedTasks[i] = taskToMark;
+        }
```

Comment:
This `for` loop looks very similar to the one in the previous method. Extract similar code blocks as utility methods.